sys/net/application_layer/dns/dns.c

```diff
@@ -95,6 +95,10 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
 
     /* skip all queries that are part of the reply */
     for (unsigned n = 0; n < ntohs(hdr->qdcount); n++) {
+        if (bufpos >= (buf + len)) {
+            /* out-of-bound */
+            return -EBADMSG;
+        }
         bufpos += _skip_hostname(bufpos);
```

Comment:
This is still open to a DOS attack, because `_skip_hostname` is very broken and will do out-of-bounds reads for many inputs. It also isn't supplied with the packet size, so it can't even detect out-of-bounds reads.

Reply:
I fixed `_skip_hostname()` to check the buffer bounds in e149b6a

Reply:
> I fixed _skip_hostname() to check the buffer bounds in e149b6a

oops there was a compile error I did not check which is fixed in f44fdcc